client/data/deposits/resolvers.js

```diff
@@ -84,7 +84,7 @@ const formatQueryFilters = ( query ) => ( {
 	],
 	status_is: query.statusIs,
 	status_is_not: query.statusIsNot,
-	locale: query.userLocale,
+	locale: query.locale,
```

Comment:
Leaving this as a generic query param, since this could be site, user or manually-selected locale in the future. These endpoints do not mandate this to be the user locale.

Reply:
(Side note not relevant to this PR) 

Is this `formatQueryFilters` operating as an "allow list"?

It looks like the outcome of this method is to:

- Convert "pascalCase" keys to underscore snake_case vibe (server uses snake_case convention).
- Format (render to consistent text) some query data e.g. dates.
- And in the case of `match` and `locale`, allow them through unchanged.

Low priority questions…

- Is the "allow list" functionality intended, would it be simpler to trust the client (calling code) and only format what needs formatting, allow any keys through?
- Could we use a generic snake_case conversion instead of doing it manually, key by key?